lib/Debugger.ProtocolHandler/RuntimeImpl.cpp

```diff
@@ -336,6 +369,16 @@ namespace JsDebug
 
                 if (GetTypeStringAndValue(arguments[i], &typeString, &objectValue))
                 {
+                    JsValueType valueType;
+                    JsGetValueType(objectValue, &valueType);
+                    if (valueType == JsValueType::JsObject) {
```

Comment:
so with this the object is represented in string and it is not expandable in the console window, right?

Reply:
Yes, it is shown as String. For object expansion, we need to send  something like

{type: "debug", args: [{type: "object", className: "Object", description: "Object",…}],…}
1.	args: [{type: "object", className: "Object", description: "Object",…}]
1.	0: {type: "object", className: "Object", description: "Object",…}
1.	className: "Object"
2.	description: "Object"
3.	objectId: "{"injectedScriptId":1,"id":91}"
4.	preview: {type: "object", description: "Object", overflow: false,…}
1.	description: "Object"
2.	overflow: false
3.	properties: [{name: "x", type: "number", value: "1"}]
1.	0: {name: "x", type: "number", value: "1"}
1.	name: "x"
2.	type: "number"
3.	value: "1"
4.	type: "object"
5.	type: "object"
2.	executionContextId: 1
3.	stackTrace: {,…}
1.	callFrames: [{functionName: "testConsoleAPIs", scriptId: "68",…}, {functionName: "", scriptId: "68",…},…]
4.	timestamp: 1557948216171.313
5.	type: "debug"

Reply:
We need to decompose the object and fill the preview field.

Reply:
what about the circular references?
obj = {};
obj.x = obj;
console.log(obj)

Reply:
JSON.stringy would fail, [Object object] will be displayed